src/components/themed/TransactionListTop.tsx

```diff
@@ -181,6 +206,180 @@ export class TransactionListTopComponent extends React.PureComponent<Props, Stat
+
+    await Airship.show(bridge => (
+      <ModalUi4
+        bridge={bridge}
+        title={sprintf(lstrings.trade_s, sceneCurrencyCode)}
+        onCancel={() => {
+          bridge.resolve()
+        }}
+      >
+        <SelectableRow
+          marginRem={0.5}
+          title={sprintf(lstrings.trade_s, sceneCurrencyCode)}
+          onPress={() => this.handleTradeSwap(bridge)}
+          icon={
+            <View style={styles.singleIconContainer}>
+              <Ionicons name="swap-horizontal" size={theme.rem(2.5)} color={theme.iconTappable} />
+            </View>
+          }
+        />
+        <SelectableRow
+          marginRem={0.5}
+          title={sprintf(lstrings.title_plugin_buy_s, sceneCurrencyCode)}
+          onPress={() => this.handleTradeBuy(bridge)}
+          icon={
+            <View style={styles.dualIconContainer}>
+              <FontAwesomeIcon name="bank" {...buySellIconProps} />
+              <AntDesignIcon name="arrowright" {...buySellIconProps} />
+            </View>
+          }
+        />
+        <SelectableRow
+          marginRem={0.5}
+          title={sprintf(lstrings.title_plugin_sell_s, sceneCurrencyCode)}
+          onPress={() => this.handleTradeSell(bridge)}
+          icon={
+            <View style={styles.dualIconContainer}>
+              <AntDesignIcon name="arrowright" {...buySellIconProps} />
+              <FontAwesomeIcon name="bank" {...buySellIconProps} />
+            </View>
+          }
+        />
+      </ModalUi4>
+    ))
+  }
+
+  handleTradeBuy = (bridge: AirshipBridge<void>) => {
+    const { navigation, wallet, tokenId } = this.props
+    const { pluginId } = wallet.currencyInfo
+
+    navigation.navigate('buyTab', { screen: 'pluginListBuy', params: { filterAsset: { pluginId, tokenId } } })
+    bridge.resolve()
+  }
+
+  handleTradeSell = (bridge: AirshipBridge<void>) => {
+    const { navigation, wallet, tokenId } = this.props
+    const { pluginId } = wallet.currencyInfo
+
+    navigation.navigate('sellTab', { screen: 'pluginListSell', params: { filterAsset: { pluginId, tokenId } } })
+    bridge.resolve()
+  }
+
+  handleTradeSwap = (bridge: AirshipBridge<void>) => {
+    const { account, navigation, wallet, tokenId } = this.props
+
+    const sceneWallet = wallet
+    const sceneTokenId = tokenId
+    const { currencyWallets } = account
+    const { rootState } = this.props
+
+    // Check balances for the displayed asset on this scene:
+    const sceneAssetCryptoBalance = wallet.balanceMap.get(tokenId)
+
+    // Filter owned assets (mainnet and token) by those defined by the
+    // SWAP_ASSET_PRIORITY:
+    const ownedAssets: Array<{ pluginId: string; tokenId: EdgeTokenId; walletId: string; dollarValue: number }> = []
+    SWAP_ASSET_PRIORITY.forEach(priorityAsset => {
+      // Filter the currencyWallets for nonzero crypto balances that are
+      // NOT the asset shown on the scene
+      Object.values(currencyWallets).forEach(accountWallet => {
```

Comment:
rename to `currencyWallet` That's the singular of `currencyWallets` which is convention.

Reply:
This made more sense in a prior implementation where I was juggling multiple wallets.